Props/Cat.gd

```diff
@@ -0,0 +1,34 @@
+extends CharacterBody2D
+
+const speed = 50
+
+var destination: Node2D
+@onready var nav_agent := $NavigationAgent2D as NavigationAgent2D
+@onready var animation_player = $AnimationPlayer
+
+func _physics_process(_delta: float) -> void:
+    var direction = to_local(nav_agent.get_next_path_position()).normalized()
+    velocity = direction * speed
```

Comment:
The `next_path_position` from the NavAgent tells us which direction to move in on the shortest path. Then we change the cat's velocity to move in that direction..